### PR TITLE
Add documentation for CSFs

### LINKED ISSUE #48: Document CSFs in the manual

### PR DESCRIPTION
**Pull request by mortenpi**
Also includes a small theory section. Still somewhat WIP, but could use another set of eyes.

Close #48.

### CODE REVIEW

File: docs/src/csfs.md [hunk 1]
```diff
@@ -1 +1,135 @@
...
+
+In AtomicLevels, we assume that the coupling is done by starting from the leftmost orbital pair, coupling those subshells together. Then the coupled two subshells are taken and coupled to the next subshells, and so on. In the end, we get a coupling tree that looks something like this:
+
+![](couplingtree.svg)
+
+The total angular momenta of the last coupling (``J_{1:k}``) determines the angular momentum of the whole CSF.
+
+## Coupling schemes
+
+Various coupling schemes exists and AtomicLevels is currently opinionated about it, using the scheme described above. The only generality is that it allows for different coupling schemes depending on whether one works with relativistis or non-relativistic orbitals.
```
Comment (jagot) [inline docs/src/csfs.md:109]: relativistis -> relativistic

**Comment by jagot** [comment]
I did not notice until now that the illustration was hand drawn...

**Comment by mortenpi** [comment]
> I did not notice until now that the illustration was hand drawn...

Just a placeholder at the moment. Should probably be replaced before merging. :stuck_out_tongue:

File: src/csfs.jl [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+Represents a single _configuration state function_ (CSF) adapted to angular momentum
+symmetries. Depending on the type parameters, it can represent both non-relativistic CSFs in
+[``LS``-coupling](@ref) and relativistic CSFs in [``jj``-coupling](@ref).
+
+A CSF is defined by the following information:
+
+* The configuration, i.e. an ordered list of subshells together with their occupations.
+* A list of intermediate coupling terms (including the seniority quantum number to label
+  states in degenerate subspaces) for each subshell in the configuration.
+* A list of coupling terms for the coupling tree. The coupling is assume to be done by first
```
Comment (jagot) [inline src/csfs.jl:13]: the coupling is assume**d**
to be done by first **coupling**

Comment (mortenpi) [reply src/csfs.jl]: Fixed in a031847.

**Comment by jagot** [comment]
Thanks!

**Pull request merged**